### PR TITLE
Add tuner callback docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Add usage docs for recently added tuner callbacks

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @Felonious-Spellfire @akihironitta

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/advanced/training_tricks.rst [hunk 1]
```diff
@@ -254,6 +278,29 @@ via ``self.learning_rate`` or ``self.lr``.
 
     trainer.tune(model)
 
+You can also customize the :class:`~pytorch_lightning.callbacks.lr_finder.LearningRateFinder` callback to run at different epochs. This feature is useful while fine-tuning models.
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/training_tricks.rst:281]: I think this deserves its own section title. For example: "Customizing the lr finder". 
Maybe put it after this section, above "Advanced GPU Optimizations"?

File: docs/source-pytorch/advanced/training_tricks.rst [hunk 2]
```diff
@@ -91,6 +91,31 @@ longer training time. Inspired by https://github.com/BlackHC/toma.
     # Find the batch size
     trainer.tune(model)
 
+
+You can also customize the :class:`~pytorch_lightning.callbacks.batch_size_finder.BatchSizeFinder` callback to run at different epochs. This feature is useful while fine-tuning models since
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/training_tricks.rst:95]: same suggestion as in my other comment :)

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/advanced/training_tricks.rst (hunk 1 above)
Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst]: good idea.. done!

File: docs/source-pytorch/advanced/training_tricks.rst (hunk 2 above)
Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst]: done!

**Review by justusschock — APPROVED** [review]
